examples/getting_started/quickstart.ipynb

```diff
@@ -0,0 +1,626 @@
+    "\n",
+    "\n",
+    "# Use asyncio.run() to execute in the main thread\n",
+    "# asyncio.run(main())\n",
+    "\n",
+    "# Use the following code if running in a Jupyter notebook\n",
+    "await main()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This code:\n",
+    "1. Defines an asynchronous function using `async def`.\n",
+    "2. Uses `asyncio.gather()` to run multiple asynchronous tasks in parallel.\n",
+    "3. Processes the results in order, but the API calls themselves are made concurrently, reducing overall execution time."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Streaming Responses\n",
+    "\n",
+    "Streaming allows you to process LLM responses in real-time. This is particularly useful for generating longer responses or providing real-time feedback to users. Let's see how we can use streaming with our capital city example:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@openai.call(\"gpt-4o-mini\", stream=True)\n",
+    "@prompt_template(\n",
+    "    \"Provide detailed information about the capital of {country}. Include facts about its history, culture, and significant landmarks.\"\n",
+    ")\n",
+    "def stream_capital_info(country: str):\n",
+    "    pass\n",
+    "\n",
+    "\n",
+    "for chunk, _ in stream_capital_info(\"France\"):\n",
+    "    print(chunk.content, end=\"\", flush=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this example:\n",
+    "\n",
+    "1. We enable streaming mode with the `stream=True` parameter in the `@openai.call` decorator.\n",
+    "2. Our prompt asks for detailed information about a capital city, which is likely to generate a longer response.\n",
+    "3. The response is returned in small chunks that can be processed immediately.\n",
+    "4. We use `end=\"\"` and `flush=True` to display the output immediately without line breaks, creating a smooth streaming effect.\n",
+    "\n",
+    "This approach has several advantages:\n",
+    "\n",
+    "1. **Responsiveness**: Users see the beginning of the response immediately, rather than waiting for the entire response to be generated.\n",
+    "2. **Handling Long Outputs**: For detailed information like this, streaming allows you to start processing or displaying the response before it's fully generated.\n",
+    "3. **Real-time Interaction**: You could potentially process the chunks as they come in, allowing for real-time analysis or formatting of the response.\n",
+    "\n",
+    "For example, you could modify the processing to format the output in real-time:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import re\n",
+    "\n",
+    "\n",
+    "def format_capital_info(chunk: str) -> str:\n",
```

Comment:
this won't actually work because the chunks may be just a single character, so you'll have to manage the content as you stream it and format more than just a single chunk to get the effect you're talking about here.

For the quickstart, this may be a bit overkill, and as I mentioned in my previous comment we may want to just keep this simple and then link to the Learn section?